hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/OmRPCLoadGenerator.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.hadoop.hdds.cli.HddsVersionProvider;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.ozone.om.protocolPB.OzoneManagerProtocolClientSideTranslatorPB;
+import org.apache.hadoop.ozone.protocol.proto.OzoneManagerProtocolProtos.EchoRPCResponse;
+import java.util.concurrent.Callable;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+/**
+ * Utility to generate RPC request to OM with or without payload.
+ */
+@Command(name = "om-rpc-load",
+        aliases = "rpcl",
+        description =
+                "Generate echo RPC request to the OM " +
+                        "with or without layload. " +
+                        "Max payload size is 2097151 KB",
+        versionProvider = HddsVersionProvider.class,
+        mixinStandardHelpOptions = true,
+        showDefaultValues = true)
+public class OmRPCLoadGenerator extends BaseFreonGenerator
+        implements Callable<Void> {
+
+  private static final int RPC_PAYLOAD_MULTIPLICATION_FACTOR = 1024;
+  private static final int MAX_SIZE_KB = 2097151;
+  private Timer timer;
+  private OzoneConfiguration configuration;
+  private OzoneManagerProtocolClientSideTranslatorPB client;
+  private byte[] payloadReq;
+  private int payloadRespSize;
+  @Option(names = {"-plrq", "--payload-req"},
+          description =
+                  "Specifies the size of payload in KB in RPC request. " +
+                          "Max size is 2097151 KB",
+          defaultValue = "0")
+  private int payloadReqSizeKB = 0;
+
+  @Option(names = {"-plrp", "--payload-resp"},
+          description =
+                  "Specifies the size of payload in KB in RPC response. " +
+                          "Max size is 2097151 KB",
+          defaultValue = "0")
+  private int payloadRespSizeKB = 0;
+  @Override
+  public Void call() throws Exception {
+    if (payloadReqSizeKB < 0 || payloadRespSizeKB < 0) {
+      throw new IllegalArgumentException(
+              "RPC request or response payload can't be negative value."
+      );
+    }
+
+    configuration = createOzoneConfiguration();
+    client = createOmClient(configuration, null);
+    init();
+    if (payloadReqSizeKB > 0) {
+      //To avoid integer overflow, we cap the payload by max 2048000 KB
+      int payloadReqSize = Math.min(
+              Math.toIntExact(payloadReqSizeKB *
+                      RPC_PAYLOAD_MULTIPLICATION_FACTOR),
+              MAX_SIZE_KB);
+      payloadReq = RandomUtils.nextBytes(payloadReqSize);
+    }
+    if (payloadRespSizeKB > 0) {
+      //To avoid integer overflow, we cap the payload by max 2048000 KB
+      payloadRespSize = Math.min(
+              Math.toIntExact(payloadRespSizeKB *
+                      RPC_PAYLOAD_MULTIPLICATION_FACTOR),
+              MAX_SIZE_KB);
+    }
+    timer = getMetrics().timer("rpc-payload");
+    try {
+      for (int i = 0; i < getThreadNo(); i++) {
+        runTests(this::sendRPCReq);
+      }
```

Comment:
`runTests` already handles threading:

```suggestion
      runTests(this::sendRPCReq);
```

Reply:
Current loop could be causing:

```
Task org.apache.hadoop.ozone.freon.BaseFreonGenerator$$Lambda$112/0x000000084018cc40@6d868997 rejected from java.util.concurrent.ThreadPoolExecutor@2c383e33[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 10]
```

and exit with error (255).

Reply:
oh! got it!! I'll remove the for-loop then! Thanks Attila.